koku/masu/database/sql/reporting_ocpazure_ocp_infrastructure_back_populate.sql

```diff
@@ -103,3 +103,106 @@ INSERT INTO {{schema | sqlsafe}}.reporting_ocpusagelineitem_daily_summary (
+    pod_limit_memory_gigabyte_hours,
+    node_capacity_cpu_cores,
+    node_capacity_cpu_core_hours,
+    node_capacity_memory_gigabytes,
+    node_capacity_memory_gigabyte_hours,
+    cluster_capacity_cpu_core_hours,
+    cluster_capacity_memory_gigabyte_hours,
+    persistentvolumeclaim_capacity_gigabyte,
+    persistentvolumeclaim_capacity_gigabyte_months,
+    volume_request_storage_gigabyte_months,
+    persistentvolumeclaim_usage_gigabyte_months,
+    raw_currency,
+    cost_category_id
+)
+SELECT uuid_generate_v4() as uuid,
+    ocp_azure.report_period_id,
+    ocp_azure.usage_start,
+    ocp_azure.usage_start,
+    ocp_azure.cluster_id,
+    ocp_azure.cluster_alias,
+    ocp_azure.namespace,
+    ocp_azure.data_source,
+    ocp_azure.node,
+    ocp_azure.persistentvolumeclaim,
+    max(ocp_azure.persistentvolume),
+    max(ocp_azure.storageclass),
+    ocp_azure.resource_id,
+    CASE WHEN ocp_azure.data_source = 'Pod'
+        THEN ocp_azure.pod_labels
+        ELSE '{}'::jsonb
+    END as pod_labels,
+    CASE WHEN ocp_azure.data_source = 'Storage'
+        THEN ocp_azure.pod_labels
+        ELSE '{}'::jsonb
+    END as volume_labels,
+    NULL as source_uuid,
+    sum(ocp_azure.pretax_cost + ocp_azure.markup_cost) AS infrastructure_raw_cost,
+    sum(ocp_azure.pod_cost + ocp_azure.project_markup_cost) AS infrastructure_project_raw_cost,
+    '{"cpu": 0.000000000, "memory": 0.000000000, "storage": 0.000000000}'::jsonb as infrastructure_usage_cost,
+    '{"cpu": 0.000000000, "memory": 0.000000000, "storage": 0.000000000}'::jsonb as supplementary_usage_cost,
+    0 as pod_usage_cpu_core_hours,
+    0 as pod_request_cpu_core_hours,
+    0 as pod_limit_cpu_core_hours,
+    0 as pod_usage_memory_gigabyte_hours,
+    0 as pod_request_memory_gigabyte_hours,
+    0 as pod_limit_memory_gigabyte_hours,
+    0 as node_capacity_cpu_cores,
+    0 as node_capacity_cpu_core_hours,
+    0 as node_capacity_memory_gigabytes,
+    0 as node_capacity_memory_gigabyte_hours,
+    0 as cluster_capacity_cpu_core_hours,
+    0 as cluster_capacity_memory_gigabyte_hours,
+    0 as persistentvolumeclaim_capacity_gigabyte,
+    0 as persistentvolumeclaim_capacity_gigabyte_months,
+    0 as volume_request_storage_gigabyte_months,
+    0 as persistentvolumeclaim_usage_gigabyte_months,
+    max(ocp_azure.currency) as raw_currency,
+    max(ocp_azure.cost_category_id) as cost_category_id
+FROM {{schema | sqlsafe}}.reporting_ocpazurecostlineitem_project_daily_summary_p AS ocp_azure
+WHERE ocp_azure.usage_start >= {{start_date}}::date
+    AND ocp_azure.usage_start <= {{end_date}}::date
+    AND ocp_azure.report_period_id = {{report_period_id | sqlsafe}}
+    AND ocp_azure.namespace = 'Storage unattributed'
+GROUP BY ocp_azure.report_period_id,
+    ocp_azure.usage_start,
+    ocp_azure.cluster_id,
+    ocp_azure.cluster_alias,
+    ocp_azure.namespace,
+    ocp_azure.data_source,
+    ocp_azure.node,
+    ocp_azure.persistentvolumeclaim,
+    ocp_azure.resource_id,
```

Comment:
Why do we need to group by resource_id here? Won't this always appear as a single line item as `storage unattributed`? Would the API ever show the individual components of the storage unattributed?

Reply:
Specifically, does the ocp-on-azure endpoint group by resource-id?

Reply:
🤷 If you believe we should remove it I can look into it.